src/main/java/com/enonic/lib/guillotine/macro/HtmlLinkProcessor.java

```diff
@@ -137,13 +136,36 @@ public String process( final String text, final String urlType, final PortalRequ
                         ImageUrlParams imageUrlParams = new ImageUrlParams().
                             type( urlType ).
                             id( id ).
-                            scale( getScale( id, imageStyle, urlParams ) ).
+                            scale( getScale( imageStyle, urlParams, null ) ).
                             filter( getFilter( imageStyle ) ).
                             portalRequest( portalRequest );
 
                         final String imageUrl = portalUrlService.imageUrl( imageUrlParams );
 
-                        processedHtml = processedHtml.replaceFirst( Pattern.quote( match ), "\"" + imageUrl + "\"" );
+                        final StringBuilder replacement = new StringBuilder( "\"" + imageUrl + "\" " );
```

Comment:
This way you swallow everything written before src:
`<img this will be gone src="..."`

Reply:
Added more tests to show that this works correct